jscomp/test/import_unnecessary.res

```diff
@@ -0,0 +1,5 @@
+// Import_lazy_component is not a pure module
+// It is supposed not to be required by dynamic import
+let a = Js.import(Import_lazy_component.a)
```

Comment:
What is tested in this example?

Reply:
```res
// import_lazy_component.res
let a = Belt.Array.map([1], x => x)
```
`import_lazy_component.res` has side effect. So, `import_unnecessary.res` module also had unnecessary `require("./import_lazy_component.js");` before this PR changes. See the test output before this PR changes affect:  https://github.com/rescript-lang/rescript-compiler/pull/6232/commits/dface2b2ea3cae7dda3edfb1dc19a24d404323f6